packages/webdriverio/src/utils/Timer.ts

```diff
@@ -29,14 +33,14 @@ class Timer {
             this._fn = () => runFnInFiberContext(fn)()
         }
 
-        const retPromise = new Promise((resolve, reject) => {
+        const retPromise = new Promise<Timer>((resolve, reject) => {
             this._resolve = resolve
             this._reject = reject
         })
 
         this.start()
 
-        return retPromise
+        return retPromise as any
```

Comment:
shouldn't this be `Promise<boolean>`?

Reply:
I still didn't figure out how to handle this return type, it complains that is missing a few properties, which are all the private properties that are not optional and the class methods. This is correct if you think that a class constructor usually returns the class itself, however a class can also return an object, which seems to be the case, but I couldn't yet find a way to handle this with TS.

Reply:
Not being able to specify return type of constructors seems to be an old discussion: https://github.com/microsoft/TypeScript/issues/11588

Reply:
One not so trivial solution, as it probably requires some good refactor in all the use cases, is to use a static factory https://2ality.com/2019/11/creating-class-instances.html#solution%3A-static-factory-method